code/datums/keybinding/human_combat.dm

```diff
@@ -190,3 +190,19 @@
 		var/obj/item/weapon/gun/rifle/m46c/COgun = held_item
 		COgun.toggle_iff(human)
 		return TRUE
+
+/datum/keybinding/human/combat/toggle_shotgun_tube
+	hotkey_keys = list("Unbound")
+	classic_keys = list("Unbound")
+	name = "toggle_shotgun_tube"
+	full_name = "Toggle Shotgun Tube"
+	keybind_signal = COMSIG_KB_HUMAN_WEAPON_SHOTGUN_TUBE
+
+/datum/keybinding/human/combat/toggle_shotgun_tube/down(client/user)
+	. = ..()
+	if(.)
+		return
+	var/mob/living/carbon/human/human = user.mob
+	var/obj/item/weapon/gun/shotgun/pump/dual_tube/held_item = human.get_held_item()
```

Comment:
You need to check if what they're holding exists and is this type

Reply:
also if client is valid

Reply:
also why all this isn't a case for other hotkeys?